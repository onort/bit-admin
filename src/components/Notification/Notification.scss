@import "../../styles/main.scss";

.container {
  position: absolute;
  width: px-to-rem(240px);
  top: calc(#{$appBar-height} + #{$spacer-md});
  right: 5%;
  z-index: 999;
  background-color: darken($light-bg, 10);
  padding: $spacer-xsm $spacer-md;
  border-radius: px-to-rem(6px);
  opacity: 0.6;
}

.default {
  box-shadow: 0 2px 4px -1px rgba(0, 0, 0, 0.2), 0 4px 5px 0 rgba(0, 0, 0, 0.14),
    0 1px 10px 0 rgba(0, 0, 0, 0.12);
}

.success {
  background-color: $success;
  color: $white;
  box-shadow: 0 2px 4px -1px rgba(40, 167, 69, 0.2),
    0 4px 5px 0 rgba(40, 167, 69, 0.14), 0 1px 10px 0 rgba(40, 167, 69, 0.12);
}

.error {
  background-color: $danger;
  color: $white;
  box-shadow: 0 2px 4px -1px rgba(220, 53, 69, 0.2),
    0 4px 5px 0 rgba(220, 53, 69, 0.14), 0 1px 10px 0 rgba(220, 53, 69, 0.12);
}
