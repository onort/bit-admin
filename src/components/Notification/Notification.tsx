import React from "react"
import cx from "classnames"

import styles from "./Notification.scss"

export type NotificationTypes = "error" | "default" | "success"

interface Props {
  className?: string
  message: string
  type?: NotificationTypes
}

// TODO: Consider adding an onClose prop and close icon
const Notification: React.SFC<Props> = props => {
  const className = cx(
    styles.container,
    { [styles[`${props.type}`]]: props.type },
    props.className
  )
  return (
    <div className={className}>
      <span className={styles.message}>{props.message}</span>
    </div>
  )
}

Notification.defaultProps = {
  type: "default"
}

export default Notification
