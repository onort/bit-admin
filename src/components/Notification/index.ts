import Notification, { NotificationTypes } from "./Notification"

export default Notification

export { NotificationTypes }
