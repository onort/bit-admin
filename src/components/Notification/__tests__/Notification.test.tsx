import React from "react"
import { shallow } from "enzyme"

import Notification from "../"

const testMessage = "Test message."
const testClassName = "testClassName"

describe("<Notification />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<Notification message={testMessage} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should have text rendered", () => {
    const wrapper = shallow(<Notification message={testMessage} />)
    expect(wrapper.find(".message").text()).toBe(testMessage)
  })

  it("should render correctly if a type prop is passed", () => {
    const wrapper = shallow(<Notification message={testMessage} type="error" />)
    expect(wrapper.find(".container").hasClass("error")).toBe(true)
  })

  it("should be able to have a custom className", () => {
    const wrapper = shallow(
      <Notification message={testMessage} className={testClassName} />
    )
    expect(wrapper.find(".container").hasClass(testClassName)).toBe(true)
  })
})
