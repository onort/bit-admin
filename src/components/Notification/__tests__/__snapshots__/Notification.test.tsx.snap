// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> should match snapshot 1`] = `
<div
  className="container default"
>
  <span
    className="message"
  >
    Test message.
  </span>
</div>
`;

exports[`<Notification /> should match snapshot 2`] = `"<div class=\\"container default\\"><span class=\\"message\\">Test message.</span></div>"`;
